### PR TITLE
fix bug: eos_threshold in S2SBeamSearcher

### PR DESCRIPTION
**Pull request by My-captain**
In my experiments, logits of <EOS> is zero when the decoding step should stop.
Absolutely, `(self.eos_threshold * max_probs)` is zero too. So the `_check_eos_threshold` return False, and the heuristic search algorithm from [Sequence-to-Sequence Speech Recognition with Time-Depth Separable Convolutions](https://arxiv.org/pdf/1904.02619.pdf) will fail.

Considering the efficiency of beam search, changing `>` to `>=` is the simplest way to fix it.

# Contribution in a nutshell
Hey, this could help our community 🌱

# Scope
* [ ] I want to get done ...
* [ ] ... and hope to also achieve ...

# Notes for reviewing (optional)
This change has these implication which might need attention over here; —how should we tackle this?

# Pre-review
* [ ] (if applicable) add an `extra_requirements.txt` file
* [ ] (if applicable) add database preparation scripts & use symlinks for nested folders (to the level of task READMEs)
* [ ] (if applicable) add a recipe test entry in the depending CSV file under: tests/recipes
* [ ] create a fresh testing environment (install SpeechBrain from cloned repo branch of this PR)
* [ ] (if applicable) run a recipe test for each yaml/your recipe dataset
* [ ] check function comments: are there docstrings w/ arguments & returns? If you're not the verbose type, put a comment every three lines of code (better: every line)
* [ ] use CI locally: `pre-commit run -a` to check linters; run `pytest tests/consistency`
* [ ] (optional) run `tests/.run-doctests.sh` & `tests/.run-unittests.sh`
* [ ] exhausted patience before clicking « Ready for review » in the merge box 🍄

---

Note: when merged, we desire to include your PR title in our contributions list, check out one of our past version releases
—https://github.com/speechbrain/speechbrain/releases/tag/v0.5.14

Tip: below, on the « Create Pull Request » use the drop-down to select: « Create Draft Pull Request » – your PR will be in draft mode until you declare it « Ready for review »

### CODE REVIEW

**Comment by Adel-Moumen** [comment]
Hi @My-captain, 

Thanks for this PR. However, I believe we should not change the operator from `>` to `>=`. Indeed, in the official paper, they are using a hard constraint, and in all of the other implementations of this heuristic, people are also using the hard constraint (see: espnet). I believe that in SpeechBrain we should always match as strictly as possible the official implementation of a research work. Thus, I prefer to close this PR. 

Thanks again for your work.
